### DISCUSSION TITLE
If I file gets big, how can I take move the function out of the create function without passing get and set every time.

### DISCUSSION THREAD

**Discussion by trufa** [pmndrs/zustand#1880] [Q&A]
I'm just wondering if there's any nice pattern I can use,

let's say I have a create function that returns:

```
...
    setMoveDataByIndex: (index, moveData) => {
        set((state) => {
          state.moveDataByIndex.set(index, moveData);
          return state;
        });
      },
...
```

If I were to move `setMoveDataByIndex` to an outside function, it'd have to look something like this:

`const setMoveDataByIndex = (get, set, index, moveData) => ...`

It's subjective, but I think it kinda looks ugly and in that case, I'd just leave it there.

Is there any way you have found that avoids passing get and set as params and being able to organize functions in say different files?

Thank you in advance!

**Comment by dbritto-dev** [comment]
@trufa let's say your store is `useStore` you can use `useStore.getState()` and `useStore.setState()`

Like this: 

```ts
const setMoveDataByIndex = (index, moveData) => {
  useStore.setState(state => {
    state.moveDataByIndex.set(index, moveData)
    return state
  })
}
```

**Reply by trufa** [reply]
That's quite a nice solution, makes sense!

Thank you, I'll wait if there are some more answers, but that pretty much solves it!

**Reply by dbritto-dev** [reply]
@trufa actually, that's the only way but keep me in the loop

**Reply by trufa** [reply]
Thank you! :)
